### DISCUSSION TITLE
How to translate Theia-exclusive labels?

### DISCUSSION THREAD

**Discussion by xcariba** [eclipse-theia/theia#10740] [Q&A] [answered]
How do I translate Theia labels if translation is not present in vscode language pack?

I understand how to add my own localizations with translations in Theia extensions, in vscode extensions and I understand that most Theia UI can be translated using vscode language pack, but there is some fields, like `Properties` label in property view that is obviously not present in vscode, since vscode does not have ropery view panel at all.

Also is there any way to change what translation id is used in code? For example for `Help` menu item Theia uses Help field ID from welcome screen, instead of menu item id, and it is either not present or changed in latest language pack. It uses very handy simplified localization method, but sometimes it misses the right translation.

Thanks!

**Comment by msujew** [comment, marked as answer]
Hi @xcariba,

I plan to merge https://github.com/eclipse-theia/theia/pull/10187 within the next days, which will enable us to generate translations for different languages automatically. We will try to add translations for all of Theia for all languages which are also supported by official vscode language packs. So, hopefully after February, downstream users of Theia shouldn't have to worry about untranslated fields anymore. 

Additionally, we recommend to use language packs which align with the current vscode API version (1.53.2). We can't really have "forward compatibility" of language packs, since that might break users which simply install language packs using the vsx-registry. It always installs the packs of the current vscode API version.

**Reply by xcariba** [reply]
This PR looks great, thanks!
